### PR TITLE
(1024) Assess - Required actions to support placement

### PR DESCRIPTION
**Pull request by richpjames**
# Context
This adds another page to the assessment flow. This time for 'Required actions to support a placement'. 
It is almost identical (except from the copy) to #485 apart from if the user answers 'Yes' to 'Are there concerns that the person poses a potentially unmanageable risk to staff or others?' in which case they need to answer questions about the discussion with the AP area manager.

[Trello card](https://trello.com/c/fuddRoqF/1024-required-actions-to-support-a-placement)

## Screenshots of UI changes
Assess -- allows me to assess an application
.png)

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/form-pages/assess/assessApplication/requiredActions/requiredActions.ts [hunk 1]
```diff
@@ -61,18 +61,7 @@ export default class RequiredActions implements TasklistPage {
...
-        [this.sections[section]]: sentenceCase(this.body[section]),
-      }
-
-      if (this.body[`${section}Comments`]) {
-        response[`${this.sections[section]} Additional comments`] = this.body[`${section}Comments`]
-      }
-
-      return response
-    }, {})
+    return generateResponsesForYesNoAndCommentsSections(this.sections, this.body)
```
Comment (pezholio) [inline server/form-pages/assess/assessApplication/requiredActions/requiredActions.ts:64]: 👍

File: server/utils/formUtils.ts [hunk 2]
```diff
@@ -125,3 +126,21 @@ export const escape = (text: string): string => {
   const escapeFilter = new nunjucks.Environment().getFilter('escape')
   return escapeFilter(text).val
 }
+
+export const generateResponsesForYesNoAndCommentsSections = (
```
Comment (pezholio) [inline server/utils/formUtils.ts:130]: I wonder if this belongs in `form-pages/utils/index.ts`?

**Review by pezholio — APPROVED** [review]
LGTM - just one question about the location of some code.

File: server/utils/formUtils.ts (hunk 2 above)
Comment (richpjames) [reply server/utils/formUtils.ts:130]: Yeah that makes sense

**Review by pezholio — APPROVED** [review]

**Pull request merged**